### ISSUE TITLE
14-bit XYB encoding incorrectly used when JxlEncoderOptionsSetLossless is set to JXL_TRUE

### ISSUE THREAD

**Issue by cagelight** [libjxl/libjxl#257]
I was asked to create this issue from the discord. libjxl produces substantially larger files than cjxl in lossless mode due to the usage of 14-bit XYB, which in addition to being larger is also not truly lossless.

Code used below:
```CPP
m_opt = JxlEncoderOptionsCreate(m_enc, nullptr);
if (quality < 0) quality = 0;
JXLEE(JxlEncoderOptionsSetDistance(m_opt, quality));
JXLEE(JxlEncoderOptionsSetLossless(m_opt, quality == 0 ? JXL_TRUE : JXL_FALSE));
JXLEE(JxlEncoderOptionsSetEffort(m_opt, 3));

static constexpr JxlPixelFormat PFMT {
	.num_channels = 3,
	.data_type = JXL_TYPE_UINT8,
	.endianness = JXL_NATIVE_ENDIAN,
	.align = 0
};

m_info.xsize = width;
m_info.ysize = height;
m_info.bits_per_sample = 8;
m_info.num_color_channels = 3;
m_info.intensity_target = 255;
m_info.orientation = JXL_ORIENT_IDENTITY;
JXLEE(JxlEncoderSetBasicInfo(m_enc, &m_info));

JxlColorEncoding color_profile;
JxlColorEncodingSetToSRGB(&color_profile, JXL_FALSE);
JXLEE(JxlEncoderSetColorEncoding(m_enc, &color_profile));
JXLEE(JxlEncoderAddImageFrame(m_opt, &PFMT, buf, width * height * 3));
JxlEncoderCloseInput(m_enc);
```
(`quality` is set to 0 in this circumstance)

**Comment by lvandeve** [comment]
Does adding the following line before JxlEncoderSetBasicInfo work?

basic_info.uses_original_profile = JXL_TRUE;

What this means:

uses_original_profile=false means the image is xyb encoded (so the encoded color isn't encoded in the original ICC profile)

uses_original_profile=true means the image is encoded in the color space given by the original ICC color profile, encoded either as RGB or as YCbCr

By default, uses_original_profile is false, since by default it's lossy with XYB color.

Since lossless encoding also supports both, uses_original_profile is a separate setting. 

If the above indeed fixes the issue, then I believe it works as intended in the code, but perhaps the documentation of JxlEncoderOptionsSetLossless can be improved to mention this

**Comment by cagelight** [comment]
I do not find this to be a satisfactory solution. it *never* makes sense to have a lossless image encoded with XYB. I don't agree with leaving this minefield in the API.

**Comment by lvandeve** [comment]
A few potential ideas, which allow to use lossless with rgb and xyb still:

1) leave as-is, but document clearly at JxlEncoderOptionsSetLossless that you should also set uses_original_profile in basic_info to true

2) have JxlEncoderOptionsSetLossless also take a JxlBasicInfo as a parameter that it modifies to set uses_original_profile to true by default

3) have some way to have a default uses_original_profile, when not specifically set, it uses xyb for lossy, rgb/ycbcr for lossless. This requires some thoughts and changes to how setting the basic info works

4) Have another JxlEncoderOption that makes it override uses_original_profile by default. But then you must think about disabling this if you do want to manually override this setting in the basic info

5) disallow using lossless with xyb encoded, however there can be use cases for this

My current favorite is 2, since it does prevent the minefield by actively requiring you to provide the JxlBasicInfo parameter

**Comment by ziemek99** [comment]
Isn't that closed by #1270 now?

**Comment by jonsneyers** [comment]
Yes, this should now not happen accidentally anymore.
